src/Compilers/CSharp/Portable/Binder/Binder_Symbols.cs

```diff
@@ -261,6 +261,7 @@ private NamespaceOrTypeOrAliasSymbolWithAnnotations BindTypeOrAliasOrKeyword(Syn
                         {
                             // Case (2)(a)(1)
                             isKeyword = false;
+                            ReportDiagnosticsIfObsolete(diagnostics, symbol, syntax, hasBaseReceiver: false);
```

Comment:
>ReportDiagnosticsIfObsolete(diagnostics, symbol, syntax, hasBaseReceiver: false); [](http://example.com/codeflow?start=28&length=81)

It looks like ```UnwrapAlias``` could have already performed this check. Are we testing scenario like that?

Reply:
It also appears that, if ```symbol``` is an alias, the error could be reported again, once consumer unwraps it again. It feels like some additional investigation in this area is necessary.

Reply:
Not tested yet, but yeah I guess the following might have duplicate diagnostics:

```csharp
using System;
using var = Legacy;

var x = new();

[Obsolete]
class Legacy
{
}
```

Will need to think more about it.

Reply:
I think I should report a diagnostic if `type` is different than `symbol`. Otherwise, `UnwrapAlias` should have already done the work.

Reply:
@AlekseyTs The duplicate diagnostic issue already exists in main branch.

![image](https://user-images.githubusercontent.com/31348972/199006126-1a191666-898c-4188-8725-c53a07f4cbce.png)

The change here doesn't add a "third" diagnostic because I'm passing the "unwrapped" alias symbol to `ReportDiagnosticsIfObsolete`.

I guess this PR should be good to go as this is an already existing problem that can be addressed separately? (I'll add the extra tests though)

Reply:
> The change here doesn't add a "third" diagnostic because I'm passing the "unwrapped" alias symbol to ```ReportDiagnosticsIfObsolete```.

I missed the fact that an alias symbol is passed to ```ReportDiagnosticsIfObsolete```. I do not think we should be doing this. Consider adjusting the code to avoid that.

Reply:
> I missed the fact that an alias symbol is passed to ReportDiagnosticsIfObsolete. I do not think we should be doing this. Consider adjusting the code to avoid that.

If I passed the `type`, I think we'll end up with the diagnostic repeated 3 times. Are you okay with passing `type` but calling `ReportDiagnosticsIfObsolete` conditionally? It should have the same behavior is passing `symbol` and calling `ReportDiagnosticsIfObsolete` unconditionally if I understand correctly.